prov/shm/src/smr.h

```diff
@@ -173,7 +175,8 @@ struct smr_cmd_ctx {
 OFI_DECLARE_FREESTACK(struct smr_rx_entry, smr_recv_fs);
 OFI_DECLARE_FREESTACK(struct smr_cmd_ctx, smr_cmd_ctx_fs);
 OFI_DECLARE_FREESTACK(struct smr_tx_entry, smr_pend_fs);
-OFI_DECLARE_FREESTACK(struct smr_sar_entry, smr_sar_fs);
+OFI_DECLARE_FREESTACK(struct smr_pend_entry, smr_sar_fs);
+OFI_DECLARE_FREESTACK(struct smr_pend_entry, smr_ipc_fs);
```

Comment:
Since these are the same type and we should only ever need rx_size number of any of them, let's move this into one freestack (ie smr_pend_fs) and insert them into the same list (ie ep->rx_pend_list) that we can process differently based on the op. I think that will save memory and simplify the progressing code paths

Reply:
The issue I see with this is that iterating through the ipc pend entries is fundamentally different from the sar entries. The completion of the async operations are external to the libfabric so it can't be gated by the signal. More over, I think it's cleaner to have separate lists for outstanding sar operations and ipc operations. They can use the same stack, but I believe the code is more straight forward if we queue them on separate lists. The processing functions of the operations is also different. I don't believe there is much of a memory overhead if we maintain the separate lists.

I'm okay with the rest of the comments. Will update.